src/pool/spawn.rs

```diff
@@ -120,6 +136,19 @@ impl<T> QueueCore<T> {
             }
         }
     }
+
+    /// Scale workers.
+    pub fn scale_workers(&self, new_thread_count: usize) -> bool {
+        if new_thread_count < self.config.min_thread_count
+            || new_thread_count > self.config.max_thread_count
+        {
+            return false;
+        }
+        self.config
+            .core_thread_count
+            .store(new_thread_count, Ordering::SeqCst);
```

Comment:
If all threads are occupied by long running tasks, then scale workers may take affect after a long time. It maybe a surprise if people try to use it to shorten pending queue length.

Reply:
I think this is expected behavior if we want to ensure scale is gracefully handled.